src/runs/execution/provider.ts

```diff
@@ -0,0 +1,255 @@
+import { ChatLLM, ChatLLMOutput } from 'bee-agent-framework/llms/chat';
+import { LLM, LLMInput } from 'bee-agent-framework/llms/llm';
+import { BaseLLMOutput, EmbeddingOptions, EmbeddingOutput } from 'bee-agent-framework/llms/base';
+import { BAMChatLLM } from 'bee-agent-framework/adapters/bam/chat';
+import { BAMLLM } from 'bee-agent-framework/adapters/bam/llm';
+import { Client as BAMClient } from '@ibm-generative-ai/node-sdk';
+import { BAMChatLLMPresetModel } from 'bee-agent-framework/adapters/bam/chatPreset';
+import { OllamaChatLLM } from 'bee-agent-framework/adapters/ollama/chat';
+import { OllamaLLM } from 'bee-agent-framework/adapters/ollama/llm';
+import { Ollama } from 'ollama';
+import { OpenAIChatLLM } from 'bee-agent-framework/adapters/openai/chat';
+import { OpenAI } from 'openai';
+import { BaseMessage } from 'bee-agent-framework/llms/primitives/message';
+import { IBMVllmChatLLM } from 'bee-agent-framework/adapters/ibm-vllm/chat';
+import { IBMvLLM } from 'bee-agent-framework/adapters/ibm-vllm/llm';
+import { Client as IBMvLLMClient } from 'bee-agent-framework/adapters/ibm-vllm/client';
+import {
+  IBMVllmChatLLMPresetModel,
+  IBMVllmModel
+} from 'bee-agent-framework/adapters/ibm-vllm/chatPreset';
+import { WatsonXChatLLM } from 'bee-agent-framework/adapters/watsonx/chat';
+import { WatsonXChatLLMPresetModel } from 'bee-agent-framework/adapters/watsonx/chatPreset';
+import { WatsonXLLM } from 'bee-agent-framework/adapters/watsonx/llm';
+
+import {
+  AI_BACKEND,
+  BAM_API_KEY,
+  IBM_VLLM_CERT_CHAIN,
+  IBM_VLLM_PRIVATE_KEY,
+  IBM_VLLM_ROOT_CERT,
+  IBM_VLLM_URL,
+  OLLAMA_URL,
+  OPENAI_API_KEY,
+  WATSONX_API_KEY,
+  WATSONX_PROJECT_ID,
+  WATSONX_REGION
+} from '@/config';
+import { AIBackend } from '@/runs/execution/constants';
+
+const MAX_NEW_TOKENS = 4096;
+
+type ChatLLMParams = { model?: string; topP?: number; temperature?: number };
+
+interface EmbeddingModel {
+  modelId: string;
+  embed(input: LLMInput[], options?: EmbeddingOptions): Promise<EmbeddingOutput>;
+}
+
+interface AIProvider<
+  ChatLLMType extends ChatLLM<ChatLLMOutput>,
+  LLMType extends LLM<BaseLLMOutput> = any
+> {
+  createChatLLM: (params?: ChatLLMParams) => ChatLLMType;
+  createCodeLLM: (params?: { model?: string }) => LLMType | void;
+  createEmbeddingModel?: (params?: { model?: string }) => EmbeddingModel;
```

Comment:
This feels like youre mixing two approaches when it comes to naming. We can either name it by usecase, than it would be:

- createAssistantBackend
- createChatBackend
- createCodeBackend
- createEmbeddingBackend

or by what it is which sort of break the abstraction

- createChatLLM
- createLLM

The former is the right approach imo as default models are tracked per usecase (e.g. chat endpoint might have different default model than assistants)